tests/run_tests.py

```diff
@@ -256,4 +286,10 @@ def _parse_command_line():
     exit_val = 0
     if ret_val.errors or ret_val.failures:
         exit_val = 1
+
+    if options.clean_temp_data:
```

Comment:
If `_run_tests` raises an exception or something goes wrong after it but before we've reached the cleanup code we'll still leak files. Please wrap `_run_tests` in `try/finally`